pontoon/base/static/js/main.js

```diff
@@ -218,20 +224,37 @@ var Pontoon = (function (my) {
         for (var i=0; i<match.length; i++) {
           replacement += ' &middot; ';
         }
-
-        return getReplacement(title, replacement);
+        return self.getPlaceableMarkup(title, replacement);
       });
 
-      /* Leading and Trailing spaces */
-      string = markup(string, /^ /gmi, 'Leading space');
-      string = markup(string, / $/gmi, 'Trailing space');
+      if (whiteSpaces) {
+        string = self.markWhiteSpaces(string);
 
-      /* Tab */
-      string = markup(string, /\t/gi, 'Tab character', '&rarr;');
+      }
+
+      return string;
+    },
+
+    /*
+     * Mark leading/trailng spaces in multiline strings (that contains newlines inside).
+     * Should be applied on the fully concatenated string, doesn't handle well substrings.
+     */
+    markWhiteSpaces: function (string) {
+      /* 'm' modifier in makes ^/$ applicable to every separate line in string, not the string as the whole. */
+      string = string.replace(
+        /^(<(ins|del)>)*( )/gmi,
+        '$1' + this.getPlaceableMarkup('Leading space', ' ')
+      );
+      string = string.replace(
+        /( )(<\/(ins|del)>)*$/gmi,
+        this.getPlaceableMarkup('Trailing space', ' ') + '$2'
+      );
```

Comment:
It would be nice to split the two rules with a blank line and add comments on top (Leading space and Trailing space) like we did below.